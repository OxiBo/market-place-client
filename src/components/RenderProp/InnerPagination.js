import React, { Component } from "react";
import { Query } from "react-apollo";
import PaginationStyles from "../styled/PaginationStyles";

export default class InnerPagination extends Component {
  state = {
    page: 1,
  };
  getPage = (n) => {
    this.setState((prevState) => ({ page: prevState.page + n }));
  };
  render() {
    const { page } = this.state;
    const {
      query,
      variableName,
      variableValue,
      perPage,
      connectionType,
    } = this.props;
    console.log("from inner component varName" + variableName);
    console.log("from inner component varValue" + variableValue);
    return (
      <Query query={query} variables={{ [variableName]: variableValue }}>
        {({ data, loading, error }) => {
          if (loading) return <p>Loading...</p>;

          const count = data[connectionType].aggregate.count;
          console.log(count);
          const pages = Math.ceil(count / perPage);

          return (
            <>
              {count === 0 ? (
                <div>No items found</div>
              ) : (
                <PaginationStyles>
                  <a
                    onClick={(e) => {
                      this.getPage(-1);
                    }}
                    aria-disabled={page <= 1}
                  >
                    ← Prev
                  </a>
                  <p>
                    Page {page} of {pages}
                  </p>
                  <p>{count} Total</p>
                  <a
                    onClick={(e) => {
                      this.getPage(1);
                    }}
                    aria-disabled={page >= pages}
                  >
                    Next →
                  </a>
                </PaginationStyles>
              )}
              {this.props.children(this.state)}
            </>
          );
        }}
      </Query>
    );
  }
}
