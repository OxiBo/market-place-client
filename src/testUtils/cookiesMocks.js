import { Cookies } from "react-cookie";
export const setMyCookies = (token, type, name, id) => {
  const cookiesMocks = new Cookies();
  cookiesMocks.set("token", token);
  cookiesMocks.set("type", type);
  cookiesMocks.set("name", name);
  cookiesMocks.set("id", id);
  return cookiesMocks;
};
