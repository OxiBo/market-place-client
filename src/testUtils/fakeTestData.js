import casual from "casual";

const fakeBuyer = () => ({
  __typename: "User",
  id: "4234",
  name: casual.name,
  email: casual.email,
  age: 21,
  image:
    "https://res.cloudinary.com/di0hg10hd/image/upload/v1597378083/sickfits/qekc7yfql14ligautxzq.jpg",
  orderItems: [],
  reviews: [],
  type: "BUYER",
});

const fakeOrderItems = () => [
  {
    __typename: "OrderItem",
    id: "123asd",
    image: "",
    count: 2,
    name: "fake order item 0",
    price: 10,
    product: {
      __typename: "Product",
      id: "456w",
      name: "fake order item 0",
      seller: {
        id: "zxcv",
        name: "EbayTest",
        __typename: "Seller",
      },
    },
    reviewed: null,
    order: {
      id: "asdfg",
      finishedAt: "2020-09-02T02:34:55.835Z",
      __typename: "Order",
    },
  },

  {
    __typename: "OrderItem",
    id: "123vbn",
    image: "",
    count: 3,
    name: "fake order item 1",
    price: 10,
    product: {
      __typename: "Product",
      id: "456",
      name: "fake order item 1",
      seller: {
        id: "zxcv123",
        name: "AmazonTest",
        __typename: "Seller",
      },
    },
    reviewed: null,
    order: {
      id: "asdfgsss",
      finishedAt: "2020-09-02T02:34:55.835Z",
      __typename: "Order",
    },
  },
];

export { fakeBuyer, fakeOrderItems };
