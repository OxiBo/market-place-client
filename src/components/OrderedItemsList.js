import React from "react";
import { Query } from "react-apollo";
import {
  ALL_USER_ORDERITEMS_PAGINATION,
  FETCH_USER_ORDERITEMS,
} from "../utils/serverOperations";
import InnerPagination from "./RenderProp/InnerPagination";
import SingleOrderItem from "./SingleOrderItem"
import Error from "./ErrorMessage"; // ???
import SmallContainer from "./styled/SmallContainer";
import { orderItemsPerPage } from "../configVars";

const OrderedItemsList = () => (
  <SmallContainer>
    <InnerPagination
      query={ALL_USER_ORDERITEMS_PAGINATION}
      variableName={"userId"}
      variableValue={"ck9x9bp5i02t90765f5wwt8mf"}
      perPage={orderItemsPerPage}
      connectionType={"orderItemsConnection"}
    >
      {({ page }) =>
        console.log(page) || (
          <Query
            query={FETCH_USER_ORDERITEMS}
            variables={{
              skip: page * orderItemsPerPage - orderItemsPerPage,
              first: orderItemsPerPage,
            }}
          >
            {({ data, error, loading }) => {
              if (loading) return <p>Loading...</p>;
              if (error) return <Error error={error} />;
             
              return data.myOrderItems.map((item) => <SingleOrderItem item={item} key={item.id}/>);
            }}
          </Query>
        )
      }
    </InnerPagination>
  </SmallContainer>
);

export default OrderedItemsList;
