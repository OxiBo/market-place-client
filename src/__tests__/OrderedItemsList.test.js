import React from "react";
import { shallow, mount } from "enzyme";
import { Link } from "react-router-dom";
import toJSON from "enzyme-to-json";
import wait from "waait";
import { act } from "react-dom/test-utils";
import { ApolloConsumer } from "react-apollo";
import { MemoryRouter, withRouter } from "react-router-dom"; // https://html.developreference.com/article/13543175/React+Router+Invariant+Violation+test+fails+with+%3CLink%3E+outside+%3CRouter%3E+%5Bduplicate%5D
// why MemoryRouter - https://github.com/enzymejs/enzyme/issues/1112#issuecomment-447643271
import OrderedItemsList from "../components/OrderedItemsList";
// import { CURRENT_USER_QUERY } from "../components/User";
// import { MockedProvider } from "react-apollo/test-utils";
import { MockedProvider } from "@apollo/react-testing";
// mocking withCookies - https://github.com/reactivestack/cookies/blob/master/packages/react-cookie/src/__tests__/withCookies-test.js
import { CookiesProvider, withCookies, Cookies } from "react-cookie";
import { setMyCookies } from "../testUtils/cookiesMocks";
import {
  FETCH_USER_ORDERITEMS,
  ALL_USER_ORDERITEMS_PAGINATION,
} from "../utils/serverOperations";
import InnerPagination from "../components/RenderProp/InnerPagination";
import { fakeOrderItems } from "../testUtils/fakeTestData";

const perPage = 2;

const fetchOrderItemsMocks = [
  {
    request: {
      query: FETCH_USER_ORDERITEMS,
      variables: {
        skip: 0,
        first: perPage,
      },
    },
    result: {
      data: {
        myOrderItems: fakeOrderItems(),
      },
    },
  },
];

const orderItemsPaginationMocks = (length) => {
  return [
    {
      request: {
        query: ALL_USER_ORDERITEMS_PAGINATION,
        variables: { userId: "4234" },
      },
      result: {
        data: {
          orderItemsConnection: {
            __typename: "aggregate",
            aggregate: {
              __typename: "count",
              count: length,
            },
          },
        },
      },
    },
  ];
};

it("displays a loading message", async () => {
  const wrapper = mount(
    <MemoryRouter>
      <MockedProvider
        mocks={[...orderItemsPaginationMocks(7), ...fetchOrderItemsMocks]}
      >
        <OrderedItemsList userId={"4234"} />
      </MockedProvider>
    </MemoryRouter>
  );

  expect(wrapper.text()).toContain("Loading...");
});

describe("<OrderItemsList/>", () => {
  let wrapper, myCookies, orderedItemsList;

  beforeEach(async () => {
    myCookies = setMyCookies(
      "token_lskdjflskdjflskdjflskdf",
      "BUYER",
      "Test",
      "lskdjflskdjflskdjflskdf"
    );

    wrapper = mount(
      <MemoryRouter>
        <MockedProvider
          mocks={[...orderItemsPaginationMocks(7), ...fetchOrderItemsMocks]}
        >
          <OrderedItemsList userId={"4234"} />
        </MockedProvider>
      </MemoryRouter>
    );

    await act(async () => wait());
    wrapper.update();
    orderedItemsList = wrapper.find("OrderedItemsList");
  });

  it("renders and matches snapshot", async () => {
    expect(
      toJSON(orderedItemsList.find("div[data-test='ordered-items-list']"))
    ).toMatchSnapshot();
  });
  it("renders a list of orderItems", async () => {
    const singleItem = wrapper.find("SingleOrderItem");
    expect(singleItem.length).toBe(perPage);
  });
});
