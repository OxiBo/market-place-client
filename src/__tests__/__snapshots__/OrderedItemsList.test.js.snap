// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderItemsList/> renders and matches snapshot 1`] = `
<div
  className="sc-AxheI fDWYuG"
  data-test="ordered-items-list"
>
  <InnerPagination
    connectionType="orderItemsConnection"
    perPage={2}
    query={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": Object {
              "kind": "Name",
              "value": "ALL_USER_ORDERITEMS_PAGINATION",
            },
            "operation": "query",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "where",
                      },
                      "value": Object {
                        "fields": Array [
                          Object {
                            "kind": "ObjectField",
                            "name": Object {
                              "kind": "Name",
                              "value": "user",
                            },
                            "value": Object {
                              "fields": Array [
                                Object {
                                  "kind": "ObjectField",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "id",
                                  },
                                  "value": Object {
                                    "kind": "Variable",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "userId",
                                    },
                                  },
                                },
                              ],
                              "kind": "ObjectValue",
                            },
                          },
                        ],
                        "kind": "ObjectValue",
                      },
                    },
                  ],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "orderItemsConnection",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "aggregate",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "count",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [
              Object {
                "defaultValue": undefined,
                "directives": Array [],
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "ID",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "userId",
                  },
                },
              },
            ],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 167,
          "start": 0,
        },
      }
    }
    variableName="userId"
    variableValue="4234"
  >
    <Query
      query={
        Object {
          "definitions": Array [
            Object {
              "directives": Array [],
              "kind": "OperationDefinition",
              "name": Object {
                "kind": "Name",
                "value": "ALL_USER_ORDERITEMS_PAGINATION",
              },
              "operation": "query",
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [
                      Object {
                        "kind": "Argument",
                        "name": Object {
                          "kind": "Name",
                          "value": "where",
                        },
                        "value": Object {
                          "fields": Array [
                            Object {
                              "kind": "ObjectField",
                              "name": Object {
                                "kind": "Name",
                                "value": "user",
                              },
                              "value": Object {
                                "fields": Array [
                                  Object {
                                    "kind": "ObjectField",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "id",
                                    },
                                    "value": Object {
                                      "kind": "Variable",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "userId",
                                      },
                                    },
                                  },
                                ],
                                "kind": "ObjectValue",
                              },
                            },
                          ],
                          "kind": "ObjectValue",
                        },
                      },
                    ],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "orderItemsConnection",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "aggregate",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "count",
                                },
                                "selectionSet": undefined,
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                ],
              },
              "variableDefinitions": Array [
                Object {
                  "defaultValue": undefined,
                  "directives": Array [],
                  "kind": "VariableDefinition",
                  "type": Object {
                    "kind": "NonNullType",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "ID",
                      },
                    },
                  },
                  "variable": Object {
                    "kind": "Variable",
                    "name": Object {
                      "kind": "Name",
                      "value": "userId",
                    },
                  },
                },
              ],
            },
          ],
          "kind": "Document",
          "loc": Object {
            "end": 167,
            "start": 0,
          },
        }
      }
      variables={
        Object {
          "userId": "4234",
        }
      }
    >
      <styled.div>
        <div
          className="sc-AxjAm ilmHZU"
        >
          <a
            aria-disabled={true}
            onClick={[Function]}
          >
            ← Prev
          </a>
          <p>
            Page 
            1
             of 
            4
          </p>
          <p>
            7
             Total
          </p>
          <a
            aria-disabled={false}
            onClick={[Function]}
          >
            Next →
          </a>
        </div>
      </styled.div>
      <Query
        query={
          Object {
            "definitions": Array [
              Object {
                "directives": Array [],
                "kind": "OperationDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "FETCH_USER_ORDERITEMS",
                },
                "operation": "query",
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [
                        Object {
                          "kind": "Argument",
                          "name": Object {
                            "kind": "Name",
                            "value": "first",
                          },
                          "value": Object {
                            "kind": "Variable",
                            "name": Object {
                              "kind": "Name",
                              "value": "first",
                            },
                          },
                        },
                        Object {
                          "kind": "Argument",
                          "name": Object {
                            "kind": "Name",
                            "value": "skip",
                          },
                          "value": Object {
                            "kind": "Variable",
                            "name": Object {
                              "kind": "Name",
                              "value": "skip",
                            },
                          },
                        },
                        Object {
                          "kind": "Argument",
                          "name": Object {
                            "kind": "Name",
                            "value": "orderBy",
                          },
                          "value": Object {
                            "kind": "EnumValue",
                            "value": "createdAt_DESC",
                          },
                        },
                      ],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "myOrderItems",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "name",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "price",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "count",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "image",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "reviewed",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "id",
                                  },
                                  "selectionSet": undefined,
                                },
                              ],
                            },
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "product",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "id",
                                  },
                                  "selectionSet": undefined,
                                },
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "name",
                                  },
                                  "selectionSet": undefined,
                                },
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "seller",
                                  },
                                  "selectionSet": Object {
                                    "kind": "SelectionSet",
                                    "selections": Array [
                                      Object {
                                        "alias": undefined,
                                        "arguments": Array [],
                                        "directives": Array [],
                                        "kind": "Field",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "id",
                                        },
                                        "selectionSet": undefined,
                                      },
                                      Object {
                                        "alias": undefined,
                                        "arguments": Array [],
                                        "directives": Array [],
                                        "kind": "Field",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "name",
                                        },
                                        "selectionSet": undefined,
                                      },
                                    ],
                                  },
                                },
                              ],
                            },
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "order",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "id",
                                  },
                                  "selectionSet": undefined,
                                },
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "finishedAt",
                                  },
                                  "selectionSet": undefined,
                                },
                              ],
                            },
                          },
                        ],
                      },
                    },
                  ],
                },
                "variableDefinitions": Array [
                  Object {
                    "defaultValue": Object {
                      "kind": "IntValue",
                      "value": "0",
                    },
                    "directives": Array [],
                    "kind": "VariableDefinition",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "Int",
                      },
                    },
                    "variable": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "skip",
                      },
                    },
                  },
                  Object {
                    "defaultValue": Object {
                      "kind": "IntValue",
                      "value": "2",
                    },
                    "directives": Array [],
                    "kind": "VariableDefinition",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "Int",
                      },
                    },
                    "variable": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "first",
                      },
                    },
                  },
                ],
              },
            ],
            "kind": "Document",
            "loc": Object {
              "end": 380,
              "start": 0,
            },
          }
        }
        variables={
          Object {
            "first": 2,
            "skip": 0,
          }
        }
      >
        <SingleOrderItem
          item={
            Object {
              "__typename": "OrderItem",
              "count": 2,
              "id": "123asd",
              "image": "",
              "name": "fake order item 0",
              "order": Object {
                "__typename": "Order",
                "finishedAt": "2020-09-02T02:34:55.835Z",
                "id": "asdfg",
              },
              "price": 10,
              "product": Object {
                "__typename": "Product",
                "id": "456w",
                "name": "fake order item 0",
                "seller": Object {
                  "__typename": "Seller",
                  "id": "zxcv",
                  "name": "EbayTest",
                },
              },
              "reviewed": null,
            }
          }
          key="123asd"
        >
          <styled.div
            data-test="single-order-item"
            key="123asd"
          >
            <div
              className="sc-AxirZ futyRa"
              data-test="single-order-item"
            >
              <img
                alt="fake order item 0"
                src="https://res.cloudinary.com/di0hg10hd/image/upload/v1594237808/sickfits/n0t4uceatjy5jdoee9p0.png"
              />
              <div>
                <div>
                  <Link
                    to="/item/456w"
                  >
                    <LinkAnchor
                      href="/item/456w"
                      navigate={[Function]}
                    >
                      <a
                        href="/item/456w"
                        onClick={[Function]}
                      >
                        <h3>
                          fake order item 0
                           
                        </h3>
                      </a>
                    </LinkAnchor>
                  </Link>
                  <hr />
                  <p>
                    Qty: 
                    <span>
                      2
                    </span>
                  </p>
                  <p>
                    Each: 
                    <span>
                      $
                      0.1
                    </span>
                  </p>
                  <p>
                    SubTotal: 
                    <span>
                      $ 
                      0.2
                    </span>
                  </p>
                  <p>
                    Sold By:
                     
                    <span>
                      <Link
                        to="/seller/zxcv"
                      >
                        <LinkAnchor
                          href="/seller/zxcv"
                          navigate={[Function]}
                        >
                          <a
                            href="/seller/zxcv"
                            onClick={[Function]}
                          >
                            EbayTest
                          </a>
                        </LinkAnchor>
                      </Link>
                    </span>
                  </p>
                </div>
                <Link
                  to={
                    Object {
                      "pathname": "/item/456w/fake order item 0/review/false",
                    }
                  }
                >
                  <LinkAnchor
                    href="/item/456w/fake order item 0/review/false"
                    navigate={[Function]}
                  >
                    <a
                      href="/item/456w/fake order item 0/review/false"
                      onClick={[Function]}
                    >
                      <styled.button>
                        <button
                          className="sc-AxiKw kVNIMs"
                        >
                          Write review
                        </button>
                      </styled.button>
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </div>
          </styled.div>
        </SingleOrderItem>
        <SingleOrderItem
          item={
            Object {
              "__typename": "OrderItem",
              "count": 3,
              "id": "123vbn",
              "image": "",
              "name": "fake order item 1",
              "order": Object {
                "__typename": "Order",
                "finishedAt": "2020-09-02T02:34:55.835Z",
                "id": "asdfgsss",
              },
              "price": 10,
              "product": Object {
                "__typename": "Product",
                "id": "456",
                "name": "fake order item 1",
                "seller": Object {
                  "__typename": "Seller",
                  "id": "zxcv123",
                  "name": "AmazonTest",
                },
              },
              "reviewed": null,
            }
          }
          key="123vbn"
        >
          <styled.div
            data-test="single-order-item"
            key="123vbn"
          >
            <div
              className="sc-AxirZ futyRa"
              data-test="single-order-item"
            >
              <img
                alt="fake order item 1"
                src="https://res.cloudinary.com/di0hg10hd/image/upload/v1594237808/sickfits/n0t4uceatjy5jdoee9p0.png"
              />
              <div>
                <div>
                  <Link
                    to="/item/456"
                  >
                    <LinkAnchor
                      href="/item/456"
                      navigate={[Function]}
                    >
                      <a
                        href="/item/456"
                        onClick={[Function]}
                      >
                        <h3>
                          fake order item 1
                           
                        </h3>
                      </a>
                    </LinkAnchor>
                  </Link>
                  <hr />
                  <p>
                    Qty: 
                    <span>
                      3
                    </span>
                  </p>
                  <p>
                    Each: 
                    <span>
                      $
                      0.1
                    </span>
                  </p>
                  <p>
                    SubTotal: 
                    <span>
                      $ 
                      0.3
                    </span>
                  </p>
                  <p>
                    Sold By:
                     
                    <span>
                      <Link
                        to="/seller/zxcv123"
                      >
                        <LinkAnchor
                          href="/seller/zxcv123"
                          navigate={[Function]}
                        >
                          <a
                            href="/seller/zxcv123"
                            onClick={[Function]}
                          >
                            AmazonTest
                          </a>
                        </LinkAnchor>
                      </Link>
                    </span>
                  </p>
                </div>
                <Link
                  to={
                    Object {
                      "pathname": "/item/456/fake order item 1/review/false",
                    }
                  }
                >
                  <LinkAnchor
                    href="/item/456/fake order item 1/review/false"
                    navigate={[Function]}
                  >
                    <a
                      href="/item/456/fake order item 1/review/false"
                      onClick={[Function]}
                    >
                      <styled.button>
                        <button
                          className="sc-AxiKw kVNIMs"
                        >
                          Write review
                        </button>
                      </styled.button>
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </div>
          </styled.div>
        </SingleOrderItem>
      </Query>
    </Query>
  </InnerPagination>
</div>
`;
